crates/store/src/state.rs

```diff
@@ -96,17 +96,114 @@ pub struct TransactionInputs {
     pub found_unauthenticated_notes: BTreeSet<NoteId>,
 }
 
+#[derive(Debug)]
+pub struct Blockchain(Mmr);
+
+impl Blockchain {
+    pub fn get(&self) -> &Mmr {
+        &self.0
+    }
+    pub fn get_mut(&mut self) -> &mut Mmr {
+        &mut self.0
+    }
+
+    pub fn get_latest_block_number(&self) -> BlockNumber {
+        let block_number: u32 = (self.0.forest() - 1)
+            .try_into()
+            .expect("chain_mmr always has, at least, the genesis block");
+
+        block_number.into()
+    }
+
+    pub fn get_chain_length(&self) -> usize {
```

Comment:
After looking at how this would be used, I think returning a `BlockNumber` is better after all, even though I find it is a bit odd to represent the _length_ of the chain as such.
But this would allow doing something like `block_chain.peaks_at(block_chain.chain_length())` which is better than requiring conversion to `usize` or some other integer type.